### DISCUSSION TITLE
show placeholder again when resetting select options

### DISCUSSION THREAD

**Discussion by RwwL** [react-hook-form/react-hook-form#11117] [General]
Working with a standard HTML `<select>` and `useController`: if the user makes a selection then changes another field which requires that the select options change, I'm finding that I can't get the `select` to show its `placeholder` again. The UI always shows the first option of the new options, even if I've called `resetField` or `setValue` first — but it's just showing in the UI, the form does have the field now set to `null` or `undefined`, so it creates a UX disconnect (user thinks a selection is made but it isn't).

For example, in a typical form with `country` and `state` field names (where the `state` can actually be province/region/whatever, switched out depending on the country).

```js
  const selectedCountry = watch('country');
  const selectedDivision = watch('state');

  useEffect(() => {
    if (!isLoadingCountries) {
      const currentCountry = countries.find((country) => country.code === selectedCountry)
      const currentDivisions = currentCountry?.divisions;
      if (currentDivisions) {
        const divisionCodes = currentDivisions.map(d => d.code);
        if (selectedDivision && !divisionCodes.includes(selectedDivision)) {
          resetField('state', { defaultValue: undefined });
          // I've also tried:
          // resetField('state', { defaultValue: null });
          // setValue('state', null);
          // setValue('state', undefined);
        }
        setDivisions(currentDivisions);
        // I've also tried those setValue/resetField attempts from above after setDivisions
      }
    }
  }, [countries, selectedCountry, selectedDivision, isLoadingCountries, resetField]);
```

Any suggestions how I can get the placeholder to show in this case?

**Comment by Moshyfawn** [comment]
[`undefined` is not a valid field value](https://react-hook-form.com/docs/useform#:~:text=You%20should%20avoid%20providing%20undefined%20as%20a%20default%20value%2C%20as%20it%20conflicts%20with%20the%20default%20state%20of%20a%20controlled%20component.).

**Reply by RwwL** [reply]
OK, thanks, that's good to know, but I've also been trying to use `null` as noted in the post, any thoughts on that (or on some other way to get to the desired result)?

**Reply by Bilal-Abubakari** [reply]
@RwwL Did you ever find a solution to this?

**Reply by RwwL** [reply]
@Bilal-Abubakari Nope, never did, unfortunately.

**Reply by Bilal-Abubakari** [reply]
@RwwL when I used empty string it worked for me
